BnmaAnalysis.R

```diff
@@ -0,0 +1,186 @@
+bnma::certolizumab
+
+
+#Binomial MetaInsight-long-formatted data
+BRdataBin <- read.csv(".\\Datasets\\BinOutBR.csv")
+
+
+#------------------------------------------
+# Convert BRdataBin to wide format
+#------------------------------------------
+
+# Clareece, this is quite crude compared to what you did. I couldn't work out how to do it with pivot_wider, or how to do it in one step using dcast.
+
+
+#' Adds a new column 'count' to 'dataFrame', which is a running total of the number
+#'  of treatments within each study
+#'
+#' @param dataFrame A data frame with the column 'Study'
+#' @return dataFrame with new column 'count'
+CountTreatment <- function(dataFrame){
+  dataFrame$count <- 1
+  for(i in 2:length(dataFrame[[1]])){
+    if(dataFrame$Study[i] == dataFrame$Study[i-1]){
+      dataFrame$count[i] <- dataFrame$count[i-1] + 1
+    }
+  }
+  return(dataFrame)
+}
+
+
+BRdataBin <- CountTreatment(BRdataBin)
+#Convert BRdataBin to wide format
+BRdataBinT <- reshape2::dcast(BRdataBin, formula=Study~count, value.var="T")
+BRdataBinR <- reshape2::dcast(BRdataBin, formula=Study~count, value.var="R")
+BRdataBinN <- reshape2::dcast(BRdataBin, formula=Study~count, value.var="N")
+BRdataBinT <- rename(BRdataBinT, "T.1"="1", "T.2"="2", "T.3"="3", "T.4"="4")
+BRdataBinR <- rename(BRdataBinR, "R.1"="1", "R.2"="2", "R.3"="3", "R.4"="4")
+BRdataBinN <- rename(BRdataBinN, "N.1"="1", "N.2"="2", "N.3"="3", "N.4"="4")
+BRdataBin_wide <- merge(merge(BRdataBinT, BRdataBinN, by="Study"), BRdataBinR, by="Study")
+
+#------------------------------------------
+
+
+#Continuous MetaInsight-long-formatted data
+BRdataCont <- read.csv(".\\Datasets\\ContOutBR.csv")
+
+
+
+
+#====================================================================
+# The functions
+#====================================================================
+
+#' Convert wide format to long format (including covariate columns)
+#' 
+#' @param wide_data Data frame of wide format
+#' @param ConBi Indicator whether outcome is binary or continuous
+#' @return Data frame in long format
+WideToLong <- function(wide_data, ConBi) {
+  # Specify columns that contain wide data
+  if (ConBi == "Continuous") {
+    change_cols <- wide_data %>%
+      select(starts_with(c("T","N","Mean","SD")))
+  } else {
+    change_cols <- wide_data %>%
+      select(starts_with(c("T","N","R")))
+  }
+  # Transform to long
+  long_data <- wide_data %>%
+    tidyr::pivot_longer(cols = names(change_cols),
+                        names_to = c(".value", "arm"),
+                        names_pattern = "(.*).(.)",
+                        values_drop_na = TRUE
+    )
+  return(long_data[,names(long_data)!="arm"])
+}
+
+
+
+
+
+#' Takes MetaInsight data and converts it into BNMA format
+#'
+#' @param BRdata A data frame of data in MetaInsight format
+#' @param dataFormat "long" or "wide"
+#' @param outcomeType "continuous" or "binomial"

```

Comment:
The options need to be capitalised, i.e. 'Continuous' or 'Binomial', as this is what the shiny input spits out when users select their option.
This relates to lines 128, 130, 152, and 163.